reconcile/dashdotdb_slo.py

```diff
@@ -132,13 +136,22 @@ def _get_service_slo(self, slo_document: SLODocumentV1) -> list[ServiceSLO]:
                 template = jinja2.Template(expr)
                 window = slo.slo_parameters.window
                 promquery = template.render({"window": window})
-                prom_response = self._promget(
-                    url=promurl,
-                    params={"query": (f"{promquery}")},
-                    token=promtoken,
-                    username=username,
-                    password=password,
-                )
+                try:
+                    prom_response = self._promget(
+                        url=promurl,
+                        params={"query": (f"{promquery}")},
+                        token=promtoken,
+                        username=username,
+                        password=password,
+                    )
+                except requests.exceptions.ConnectionError as error:
+                    # This can happen when prometheus is unreachble, or when running locally
+                    # and some prometheus URL are openshift service names. The trick is to run
+                    # with `oc port-forward` and update the local hosts file if we need to query those.
+                    LOG.error(
+                        f"{self.logmarker} Could not reach prometheus at {promurl}: {error}. Skipping {slo.name}"
+                    )
+                    continue
```

Comment:
I don't like this. We are not going to know when things are breaking since no one is watching logs. And this is not how our integrations work. For better or worse, a break in a single element always make the whole integration fail. We should not introduce new patterns unless we can alert on them.

Reply:
agreed, fixed in https://github.com/app-sre/qontract-reconcile/pull/4713/commits/6b409c8eac0feb11c17b28eab8bbd1ed539a7235